### ISSUE TITLE
zabbix_template creates template in check mode

### ISSUE THREAD

**Issue by chemnic** [ansible-collections/community.zabbix#49]
##### SUMMARY

The module zabbix_template creates a new template when in check mode with zabbix 4.0.
I use template_json to import a new template with just the enough data to create it.
The normal mode (not in check mode) works as expected: Shows "changed" when the template doesn't exist, and just "ok" when it already exists.
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

zabbix_template
##### ANSIBLE VERSION

```paste below
ansible 2.9.9
  config file = /home/chestrada/ansible/vfmsa_ansible/ansible.cfg
  configured module search path = ['/home/chestrada/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/chestrada/xxx/ansible/lib/python3.6/site-packages/ansible
  executable location = /home/chestrada/xxx/ansible/bin/ansible
  python version = 3.6.9 (default, Apr 18 2020, 01:56:04) [GCC 8.4.0]

```

##### CONFIGURATION

```paste below
ANSIBLE_PIPELINING(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = -o ControlMaster=auto -o ControlPersist=1800s -o PreferredAuthentications=publickey
CACHE_PLUGIN(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = ~/ansible_cache
CACHE_PLUGIN_TIMEOUT(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = 86400
DEFAULT_CALLBACK_PLUGIN_PATH(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = ['/home/chestrada/py_ansible/lib/python3.6/site-packages/ara/plugins/callba
DEFAULT_FORKS(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = 50
DEFAULT_GATHERING(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = smart
DEFAULT_INTERNAL_POLL_INTERVAL(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = 0.001
DEFAULT_ROLES_PATH(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = ['/home/chestrada/ansible/vfmsa_roles']
DEFAULT_STDOUT_CALLBACK(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = yaml
DEFAULT_VAULT_PASSWORD_FILE(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = /home/chestrada/.ansible_vault
RETRY_FILES_SAVE_PATH(/home/chestrada/ansible/vfmsa_ansible/ansible.cfg) = /home/chestrada/.ansible-retry

```

##### OS / ENVIRONMENT

Ubuntu 18.04
Zabbix 4.0
##### STEPS TO REPRODUCE

Run the playbook in check mode when the template still doesn't exist:
```
ansible-playbook -i inventory.yml test_playbook.yml -l srvd-test06 --check

PLAY [all] **************************************************************************************************************************************************

TASK [include_vars - Load vars form file] *******************************************************************************************************************
ok: [srvd-test06]

TASK [Import template] **************************************************************************************************************************************
changed: [srvd-test06 -> localhost]

PLAY RECAP **************************************************************************************************************************************************
srvd-test06                : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 


```

test_playbook.yml:
```yaml
---
- hosts: all
  tasks:

  - name: "include_vars - Load vars form file"
    include_vars: {file: testing_vars.yml, name: varsfromfile}

  - name: "Import template"
    zabbix_template:
      state: present
      validate_certs: false
      server_url: "https://{{ ansible_host }}"
      login_user: Admin
      login_password: "{{ zabbix_admin_pass | default('zabbix') }}"
      template_json: "{{ varsfromfile }}"
    delegate_to: localhost
...
```
testing_vars.yml:
```yaml
---
zabbix_export:
  version: '4.0'
  templates:
    - name: templates_name_test1
      template: templates_template_test1
      description: templates_description_test1
      groups:
        - name: Templates
...
```

##### EXPECTED RESULTS

Ansible should only check for the template and report if there is going to be change. But not make changes.

##### ACTUAL RESULTS

Ansible creates the template, which can be confirmed in the Zabbix GUI.

```paste below

```

**Comment by chemnic** [comment]
Did someone managed to reproduced this? Is this really a bug or there is something wrong I'm doing?

**Comment by D3DeFi** [comment]
@chemnic sorry, haven't had time to look at this yet. Thank you for opening the issue and letting us know! 

I will try to reproduce it as soon as possible and let you know

**Comment by D3DeFi** [comment]
@chemnic it indeed was a bug :)

**Comment by chemnic** [comment]
It looks fixed. I think.
The template is no longer created in check mode.
But after being created in normal mode, it reports 'changed' in check mode (not idempotent), and reports 'ok' in normal mode (idempotent).

Is this related with the same issue?

**Comment by D3DeFi** [comment]
@chemnic seems that you've hit bug from waaay back :+1: Thanks for thoroughly testing this :)

Fix in #69

**Comment by chemnic** [comment]
Ok. Its fixed. Thank you for the hard work.
